logstash-core/spec/logstash/instrument/wrapped_write_client_spec.rb

```diff
@@ -18,39 +18,54 @@
 
   subject { described_class.new(write_client, pipeline, metric, plugin) }
 
+  def threaded_read_client(read_client)
+    Thread.new(read_client) do |_read_client|
```

Comment:
If we don't pass it we will need to make sure that `read_client` is create with `let!` to make sure we don't create lazy. `Thread.new` will  make a copy of any variable passed.